com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/cache/test/ResourceCacheKeyTest.java

```diff
@@ -97,103 +101,233 @@ public void tearDown() throws IOException {
 		Task task2 = builder2.addInput(input).addOutput(output).create(null);
-		String key2 = ResourceCacheKey.calculate(task2, createEmptyOptions(), output);
+		String key2 = ResourceCacheKey.calculate(task2.calculateSignature(), output);
 
 		assertNotEquals(key1, key2);
 	}
 
 	// are output resource names taken into account and produce different keys?
 	@Test
-	public void testOutputNames() throws CompileExceptionError, IOException {
+	public void testOutputNames() throws CompileExceptionError, IOException, NoSuchAlgorithmException {
 		IResource input = createResource("someInput");
 
 		DummyBuilder builder1 = new DummyBuilder();
 		IResource output1 = createResource("someOutput").output();
 		Task task1 = builder1.addInput(input).addOutput(output1).create(null);
-		String key1 = ResourceCacheKey.calculate(task1, createEmptyOptions(), output1);
+		String key1 = ResourceCacheKey.calculate(task1.calculateSignature(), output1);
 
 		DummyBuilder builder2 = new DummyBuilder();
 		IResource output2 = createResource("someOutput").output();
 		Task task2 = builder2.addInput(input).addOutput(output2).create(null);
-		String key2 = ResourceCacheKey.calculate(task2, createEmptyOptions(), output2);
+		String key2 = ResourceCacheKey.calculate(task2.calculateSignature(), output2);
 
 		assertNotEquals(key1, key2);
 	}
 
 	// some project options have no impact on key creation
 	@Test
-	public void testIgnoredOptions() throws CompileExceptionError, IOException {
+	public void testIgnoredOptions() throws CompileExceptionError, IOException, NoSuchAlgorithmException {
 		IResource input = createResource("someInput");
 		IResource output = createResource("someOutput").output();
-
-		Map<String, String> ignoredOptions1 = new HashMap<String, String>();
-		ignoredOptions1.put("email", "foo@bar.com");
-		ignoredOptions1.put("debug", "true");
+		DummyBuilder.addParamsDigest(DummyBuilder.class, Map.ofEntries(
+				entry("not_important_option", "yep")
+		), DummyBuilder.class.getAnnotation(BuilderParams.class));
 		DummyBuilder builder1 = new DummyBuilder();
 		Task task1 = builder1.addInput(input).addOutput(output).create(null);
-		String key1 = ResourceCacheKey.calculate(task1, ignoredOptions1, output);
+		String key1 = ResourceCacheKey.calculate(task1.calculateSignature(), output);
 
-		Map<String, String> ignoredOptions2 = new HashMap<String, String>();
-		ignoredOptions2.put("email", "bob@acme.com");
-		ignoredOptions2.put("debug", "false");
 		DummyBuilder builder2 = new DummyBuilder();
 		Task task2 = builder2.addInput(input).addOutput(output).create(null);
-		String key2 = ResourceCacheKey.calculate(task2, ignoredOptions2, output);
+		String key2 = ResourceCacheKey.calculate(task2.calculateSignature(), output);
 
 		assertEquals(key1, key2);
 	}
+
+	// Ensure that all parameters specified in all builders exist in Bob
+	@Test
+	public void testAllParametersExist() throws IOException, CompileExceptionError, NoSuchFieldException, IllegalAccessException {
```

Comment:
We use strings for parameters, it's useful to test if we didn't typo and all the parameters really exist